jenkinsfiles/RHEL_Check.groovy

```diff
@@ -0,0 +1,371 @@
+                        [key: 'PR_number', value: '$.number', defaultValue: 'None'],
+                        [key: 'Repository', value: '$.pull_request.base.repo.full_name', defaultValue: 'None'],
+                        [key: 'User', value: '$.pull_request.user.login', defaultValue: 'None'],
+                        [key: 'action', value: '$.action', defaultValue: 'None']
+                ],
+
+                causeString: 'Triggered on $PR_number',
+
+                token: 'oneDPL-pre-ci',
+
+                printContributedVariables: true,
+                printPostContent: true,
+
+                silentResponse: false,
+
+                regexpFilterText: '$action',
+                regexpFilterExpression: '(opened|reopened|synchronize)'
+        )
+    }
+
+    stages {
+        stage('Check_User_in_Org') {
+            agent {
+                label "master"
+            }
+            steps {
+                script {
+                    try {
+                        retry(2) {
+                            fill_task_name_description()
+                            def check_user_return = sh(script: "python3 /localdisk2/oneDPL_CI/check_user_in_group.py -u  ${env.User}", returnStatus: true, label: "Check User in Group")
+                            echo "check_user_return value is $check_user_return"
+                            if (check_user_return == 0) {
+                                user_in_github_group = true
+                            }
+                            else {
+                                user_in_github_group = false
+                                currentBuild.result = 'UNSTABLE'
+                            }
+                        }
+                    }
+                    catch (e) {
+                        fail_stage = fail_stage + "    " + "Check_User_in_Org"
+                        user_in_github_group = false
+                        echo "Exception occurred when check User:${env.User} in group. Will skip build this time"
+                        sh script: "exit -1", label: "Set Failure"
+                    }
+                }
+            }
+        }
+
+        stage('RHEL_Check') {
+            when {
+                expression { user_in_github_group }
+            }
+            agent { label "Debug_RHEL" }
+            stages {
+                stage('Git-monorepo'){
+                    steps {
+                        script {
+                            try {
+                                retry(2) {
+                                    githubStatus.setPending(this, "Jenkins/RHEL_Check")
+                                    if (fileExists('./src')) {
+                                        sh script: 'rm -rf src', label: "Remove Src Folder"
+                                    }
+
+                                    sh script: 'cp -rf /export/users/oneDPL_CI/oneDPL-src/src ./', label: "Copy src Folder"
```

Comment:
Could you please explain the purpose of copying repo from some local folder? Is it possible just clone the repo from github directly? It will look more transparently.

Reply:
This is what we discussed before, some network slowness issue may make the process waste more time in "git clone" other than copy + git pull. So that is why we currently prefer to use this way to download source code.

Reply:
Have you considered cloning with `--depth 1`? It should be much quicker than full cloning in case of big history.

Reply:
Hi, "depth 1" is a good way to save download time indeed, but cannot fulfill our CI requirement. Here is the reason.
Currently we need to download source code from oneDPL repo with designate PR. And in order to make sure the CI can verify whole scale behavior, we need to merge the PR commit with default branch, release_oneDPL. But considering the case that, may be the commit of a PR was based on a SHA from release_oneDPL several days ago. Then it will fail if we just git clone depth 1 from the source code ---- git merge cannot locate the same history point between release_oneDPL and the PR commit. An example is what I just tested based on a fork repo: http://134.191.240.166:8010/blue/organizations/jenkins/Test_Jobs%2FUB20_Check/detail/UB20_Check/72/pipeline

Another concern is when we download source code for oneAPI-samples. This repo contains many files which will cost at least half an hour to git clone even we use "depth 1" on the local machines. An example is also from a fork repo CI test:
http://134.191.240.166:8010/job/Test_Jobs/job/UB1804_Check/41/console
Comparing with git clone, local files operations will only cost 1~2 minutes. So I suggest to keep using our current method. When the network issue has been solved, we can switch to use git clone later.

Reply:
Thank you for the explanation! Then I have no more concerns for now.